var/spack/repos/builtin/packages/dealii/package.py

```diff
@@ -148,6 +148,14 @@ class Dealii(CMakePackage, CudaPackage):
     depends_on('slepc@:3.6.3',     when='@:8.4.1+slepc+petsc+mpi')
     depends_on('slepc~arpack',     when='+slepc+petsc+mpi+int64')
     depends_on('sundials@:3~pthread', when='@9.0:+sundials')
+    # Both Trilinos and SymEngine bundle the Teuchos RCP library.
+    # This leads to conflicts between macros defined in the included
+    # headers when they are not compiled in the same mode.
+    # See https://github.com/symengine/symengine/issues/1516
+    # FIXME: uncomment when the following is fixed
+    # https://github.com/spack/spack/issues/11160
+    # depends_on("symengine@0.4: build_type=Release", when="@9.1:+symengine+trilinos^trilinos~debug")
```

Comment:
The only workaround that comes to my mind is to force `symengine` to depend on `trilinos+debug` when `build_type=Debug` and on `trilinos~debug` in the other cases. Would that make sense?

Reply:
> Would that make sense?

I don't think so. They don't link against Trilinos anyhow. We just have to _synchronize_ the two in case both are used within the same software like `dealii`.

Reply:
but thanks for thinking about the issue, appreciate it 👍

Reply:
Hmm, would it make sense then to have them coordinated with the `build_type` of `dealii`?
```python
depends_on('symengine@0.4:', when='@9.1:+symengine)
depends_on('symengine build_type=Release', when='+symengine build_type=Release')
depends_on('symengine build_type=Debug', when='+symengine build_type=Debug')
depends_on('trilinos~debug', when='+trilinos build_type=Release')
depends_on('trilinos+debug', when='+trilinos build_type=Debug')
```

Reply:
> would it make sense then to have them coordinated with the build_type of dealii?

Unfortunately that's also not a solution. `deal.II` build both debug and release libraries by default (that's why our build type by default is `DebugRelease`), regardless of its dependencies. This is convenient because you can develop in `debug` mode by linking to debug library and by flipping `cmake` flags in user project switch to linking against release lib. 

Ideally this should really by coordinated only between `symengine` and `trilinos`.

p.s. Given that @jppelteret approved this PR, I think we should merge as-is.

Reply:
Since these both default to release mode, I think that disabling this is OK for now. I've had problems when working with an out-of-Spack build of SymEngine, so at least documenting that this issue exists is better than nothing at all.